src/Psy/ColorMode.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of Psy Shell.
+ *
+ * (c) 2012-2015 Justin Hileman
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Psy;
+
+/**
+ * A configuration object that represents one of 3 available color modes:
+ *   1. "auto" (the default): if color support is detected, output contains colors
+ *   2. "forced": force colors in output
+ *   3. "disabled": disable colors in output
+ */
+class ColorMode
```

Comment:
> Convince me?

I'm happy to try! If unsuccessful, constants and equality it is. :+1: A few design reasons to consider an alternative, though:

### Type hinting

Because color mode is a class, type hinting can be used to help prevent bugs (and aid in debugging existing ones).

#### Example: class and functions

If an invalid value is assigned, a clear error happens immediately:

```php
(new \Psy\Configuration)->setColorMode('oops');
```
```
PHP error:  Argument 1 passed to Psy\Configuration::setColorMode() must be an instance of Psy\ColorMode, string given, called in...
```

#### Example: constants and equality

If an invalid value is assigned, nothing happens immediately:

```php
(new \Psy\Configuration)->setColorMode('oops'); // => null
```

However, a less clear error might happen later on (ex. `call on a non-object`).

### Silent failures

Equality can lead to silent failures, since nearly anything can be compared to anything.

#### Example: class and functions

For example, imagine that the `colorMode` has not been assigned due to an initialization bug, so it's `null`:

```php
$config->colorMode(); // => null

if ($config->colorMode()->isAuto()) {
    // ...
}
```

A fairly clear error happens immediately:

```
PHP Fatal error:  Call to a member function isAuto() on a non-object...
```

#### Example: constants and equality

However, imagine the same scenario as above, but with constants and equality:

```php
$config->colorMode(); // => null

if ($config->colorMode() === \Psy::COLOR_MODE_AUTO) {
    // ...
}
```

No error occurs. However, a less clear error might happen later on.

### Skipped use of constants

Developers are sometimes tempted (myself included) to simply skip using the constants (ex. to keep their code terse). This can lead to unintended bugs:

```php
if ($config->colorMode() === 'automatic') {
    // (will never execute since "automatic" is not a valid setting)
}
```

### Unintented operators

Since equality is not a concrete, public interface, developers may be tempted to use other operators the author didn't intend. For example, `==` instead of `===`, or worse:

```php
if (preg_match('/auto/', $config->colorMode())) {
    // ...
}
```

There are a couple of other reasons (difficult to mock in tests, bound to static behavior), but I think these are the most complelling! Thanks for considering it!